### PR TITLE
fix(ingestion) Removing python restriction

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/setup.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def get_long_description():
 base_requirements = {
     # Compatability.
     "dataclasses>=0.6; python_version < '3.7'",
-    "typing_extensions>=3.10.0.2,<4",
+    "typing_extensions>=3.10.0.2",
```
Comment (anshbansal) [inline metadata-ingestion/setup.py:26]: You may want to cross-check this. I think this restriction was added for google cloud composer compatibility when one of the customers said our airflow package was not installing

Comment (treff7es) [reply metadata-ingestion/setup.py:26]: I checked before and it seems like the latest composer uses the same typing-> https://cloud.google.com/composer/docs/concepts/versioning/composer-versions

Comment (anshbansal) [reply metadata-ingestion/setup.py:26]: We really need a way to have different releases for different airflow versions. Someone might still be using an older airflow version and we would be back to compatibility issues. 
But yeah that can be done separately.

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**